stl/inc/compare

```diff
@@ -27,8 +27,25 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
-using _Literal_zero = decltype(nullptr);
-using _Compare_t    = signed char;
+struct partial_ordering;
+struct weak_ordering;
+struct strong_ordering;
+
+void _Literal_zero_is_expected();
+
+struct _Literal_zero {
+
+    consteval _Literal_zero(int _Zero) noexcept {
+        // Can't use _STL_VERIFY because this is a core header
+        if (_Zero != 0) {
+            _Literal_zero_is_expected();
+        }
+    }
+
+    template <class _Ty, class = enable_if_t<!_Is_any_of_v<_Ty, int, partial_ordering, weak_ordering, strong_ordering>>>
+    _Literal_zero(_Ty) = delete;
+};
+using _Compare_t = signed char;

```

Comment:
Suggestion: Now that `_Literal_zero` is a struct, adding a newline before `_Compare_t` would be nice.